gnd/src/main/java/com/google/android/gnd/ui/map/gms/GoogleMapsMapAdapter.java

```diff
@@ -167,6 +186,40 @@ private static LatLng toLatLng(Point point) {
     return new LatLng(point.getLatitude(), point.getLongitude());
   }
 
+  // Handle taps on ambiguous features.
+  private void handleAmbiguity(MapGeoJson feature) {
+    List<S2Loop> loops = geoJsonPolygonLoops.get(feature);
+    Collection<MapFeature> candidates = new ArrayList<>();
+
+    candidates.add(feature);
+
+    Timber.i("Handling loops: %s", loops);
+
+    if (loops == null) {
+      featureClicks.onNext(stream(candidates).collect(toImmutableList()));
+      return;
+    }
+
+    for (Marker marker : markers.getMarkers()) {
+      if (stream(loops).anyMatch(loop -> loop.contains(markerToS2Point(marker)))) {
+        candidates.add((MapFeature) marker.getTag());
+      }
+    }
```

Comment:
I think if the click was received by the map and not the marker, then we can assume the user wasn't trying to tap the marker. Is that right?